src/Moq/Setup.cs

```diff
@@ -35,7 +35,25 @@ protected Setup(InvocationShape expectation)
 
 		public MethodInfo Method => this.expectation.Method;
 
-		public abstract void Execute(Invocation invocation);
+		public bool WasMatched => (this.flags & Flags.Matched) != 0;
+
+		public void Execute(Invocation invocation)
+		{
+			// update this setup:
+			this.flags |= Flags.Matched;
+
+			// update invocation:
+			invocation.MarkAsMatchedBy(this);
+			this.SetOutParameters(invocation);
+
+			// update condition (important for `MockSequence`) and matchers (important for `Capture`):
+			this.Condition?.SetupEvaluatedSuccessfully();
```

Comment:
Interesting to note that `Condition` no longer needs to remain `public`; it is now only ever queried from within the same type.